File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder sb = new StringBuilder();
        char pre = '1', cur = 'a';
        int len = s.length(), count = 0;

        for(int i = 0; i < len; i++) {
            cur = s.charAt(i);
            if(cur == pre) {
                count++;
            }
            else {
                count = 0;
                pre = cur;
            }
            
            if(count < 2) {
                sb.append(cur);
            }
        }

        return sb.toString();
    }
}

File: 1302-delete-characters-to-make-fancy-string/1302-delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        answer = ""
        count = 1
        before = '0'
        for i in s:
            if i == before :
                count += 1
            else :
                count = 1
                before = i
            
            if count <= 2 :
                answer += i

        return answer